Handle truncate_at_block during an IF transition

This is different failure from the one reported by https://github.com/AntelopeIO/spring/issues/1365.
It happened after the fix of https://github.com/AntelopeIO/spring/issues/1365 had been merged into `sync_call` branch.

https://github.com/AntelopeIO/spring/actions/runs/14738672506/job/41371676600?pr=1466

```
Traceback (most recent call last):
  File "/__w/spring/spring/build/tests/nodeos_read_terminate_at_block_test.py", line 363, in <module>
    assert cluster.getNode(nodeId).waitForLibToAdvance(), f"LIB did not advance for {nodeId}"
AssertionError: LIB did not advance for 1
```

--- a/libraries/chain/controller.cpp
+++ b/libraries/chain/controller.cpp
@@ -2146,13 +2146,18 @@ struct controller_impl {
 
       if (conf.truncate_at_block > 0 && chain_head.is_valid()) {
          if (chain_head.block_num() == conf.truncate_at_block && fork_db_has_root()) {
-            fork_db_.apply<void>([&](auto& fork_db) {
-               if (auto head = fork_db.head(); head && head->block_num() > conf.truncate_at_block) {
-                  ilog("Removing blocks past truncate-at-block ${t} from fork database with head at ${h}",
-                        ("t", conf.truncate_at_block)("h", head->block_num()));
-                  fork_db.remove(conf.truncate_at_block + 1);
-               }
-            });
+            if (fork_db_.version_in_use() == fork_database::in_use_t::both) {
+               // in savanna transition
+               wlog("In the middle of Savanna transition, truncate-at-block not allowed, ignoring truncate-at-block ${b}", ("b", conf.truncate_at_block));
+            } else {
+               fork_db_.apply<void>([&](auto& fork_db) {
+                  if (auto head = fork_db.head(); head && head->block_num() > conf.truncate_at_block) {
+                     ilog("Removing blocks past truncate-at-block ${t} from fork database with head at ${h}",
+                           ("t", conf.truncate_at_block)("h", head->block_num()));
+                     fork_db.remove(conf.truncate_at_block + 1);
+                  }
+               });
+            }
          }
       }
 

